### PR TITLE
edx-platform compatibility fixes and test fixes

### PR DESCRIPTION
**Pull request by rehan99000**
**Description:**
This PR fixes python/django compatibility issues in various apps and updates failing tests.

*Note:*
Some changes are fixes to the errors in conflict resolution as well.

### CODE REVIEW

File: lms/djangoapps/instructor/views/api_urls.py [hunk 1]
```diff
@@ -81,9 +81,6 @@
     # Cohort management
     url(r'^add_users_to_cohorts$', api.add_users_to_cohorts, name='add_users_to_cohorts'),
 
-    # Bulk enrollment management
-    url(r'enroll_users_to_course$', api.bulk_enroll_users_to_course, name='bulk_enroll_users_to_course'),
```
Comment (zubair-arbi) [inline lms/djangoapps/instructor/views/api_urls.py:85]: Why we needed to remove the bulk enrollment feature? 
https://github.com/edly-io/edx-platform/pull/38/

File: common/djangoapps/student/tests/factories.py [hunk 2]
```diff
@@ -98,6 +98,18 @@ def profile(obj, create, extracted, **kwargs):  # pylint: disable=unused-argumen
...
 
+    @factory.post_generation
+    def edly_profile(obj, create, extracted, **kwargs):  # pylint: disable=unused-argument, missing-function-docstring
+        if create:
+            from openedx.features.edly.tests.factories import EdlyUserProfileFactory
+
+            obj.save()
+            return EdlyUserProfileFactory.create(user=obj, **kwargs)
+        elif kwargs:
+            raise Exception("Cannot build a user profile without saving the user")
```
Comment (zubair-arbi) [inline common/djangoapps/student/tests/factories.py:109]: nit: use single quote in code

File: lms/envs/test.py [hunk 3]
```diff
@@ -339,6 +339,9 @@
     ("utm_medium", "Footer"),
 ])
 
+MIDDLEWARE.remove('openedx.features.edly.middleware.EdlyOrganizationAccessMiddleware')
+MIDDLEWARE.remove('openedx.features.edly.middleware.SettingsOverrideMiddleware')
```
Comment (zubair-arbi) [inline lms/envs/test.py:343]: I think the edX test should run fine with`SettingsOverrideMiddleware`. If this is the case then please remove it from here.

File: openedx/core/djangoapps/user_authn/tests/test_cookies.py [hunk 4]
```diff
@@ -1,5 +1,6 @@
 # pylint: disable=missing-docstring
 
+
```
Comment (zubair-arbi) [inline openedx/core/djangoapps/user_authn/tests/test_cookies.py:3]: remove this extra line

File: openedx/features/edly/tests/utils.py [hunk 5]
```diff
@@ -0,0 +1,30 @@
+from django.conf import settings
+from django.test.client import RequestFactory
+
+from rest_framework.test import APIClient
+
+from openedx.core.djangoapps.site_configuration.tests.factories import SiteFactory
+from openedx.features.edly import cookies
+from openedx.features.edly.tests.factories import EdlySubOrganizationFactory
+
+def get_edly_middleware_authorized_client(enforce_csrf_checks=False, domain=None, client_class=None):
```
Comment (zubair-arbi) [inline openedx/features/edly/tests/utils.py:10]: I am not seeing the usage of this method. Can we remove it?

Also the name can be simplified like`get_edly_client`

**Review by zubair-arbi — CHANGES_REQUESTED** [review]
I have added my initial feedback.

File: lms/envs/test.py (hunk 3 above)
Comment (rehan99000) [reply lms/envs/test.py:343]: some tests had issues with it. its better to not use this with the tests that we haven't written i think

File: common/djangoapps/student/tests/factories.py (hunk 2 above)
Comment (rehan99000) [reply common/djangoapps/student/tests/factories.py]: done

File: lms/djangoapps/instructor/views/api_urls.py (hunk 1 above)
Comment (rehan99000) [reply lms/djangoapps/instructor/views/api_urls.py]: added back

File: openedx/core/djangoapps/user_authn/tests/test_cookies.py (hunk 4 above)
Comment (rehan99000) [reply openedx/core/djangoapps/user_authn/tests/test_cookies.py]: removed

File: openedx/features/edly/tests/utils.py (hunk 5 above)
Comment (rehan99000) [reply openedx/features/edly/tests/utils.py]: removed

File: lms/djangoapps/commerce/api/v1/tests/test_views.py [hunk 6]
```diff
@@ -175,7 +175,7 @@ def test_update(self):
         self.assertIsNone(VerificationDeadline.deadline_for_course(self.course.id))
 
         # Generate the expected data
-        verification_deadline = datetime(year=2020, month=12, day=31, tzinfo=pytz.utc)
+        verification_deadline = datetime(year=2021, month=12, day=31, tzinfo=pytz.utc)
```
Comment (iam-mhaseeb) [inline lms/djangoapps/commerce/api/v1/tests/test_views.py:178]: This would fail again in 2022. One way is to get the current year and add 1 year in it.

File: lms/djangoapps/discussion/django_comment_client/base/tests.py [hunk 7]
```diff
@@ -1962,7 +1962,7 @@ def _raise_navigation_event(self, label, include_name):
             data=self.create_segmentio_event_json(**kwargs),
             content_type='application/json',
         )
-        User.objects.create(pk=SEGMENTIO_TEST_USER_ID, username=str(mock.sentinel.username))
+        User.objects.create(username=str(mock.sentinel.username))
```
Comment (iam-mhaseeb) [inline lms/djangoapps/discussion/django_comment_client/base/tests.py:1965]: Won't this create a user with random PK? If some tests are depending on that specific pk then it would fail.

File: lms/templates/instructor/instructor_dashboard_2/membership.html [hunk 8]
```diff
@@ -68,39 +68,6 @@
     <div class="request-response-error"></div>
 </fieldset>
 
-<!-- Uploading a CSV file of bulk enrollment. -->
```
Comment (iam-mhaseeb) [inline lms/templates/instructor/instructor_dashboard_2/membership.html:71]: Why we are removing the bulk enrollment feature?

File: openedx/core/djangoapps/user_authn/tests/test_cookies.py [hunk 9]
```diff
@@ -115,18 +115,15 @@ def test_set_logged_in_jwt_cookies(self):
...
-            with patch.dict("django.conf.settings.JWT_AUTH", {"JWT_PRIVATE_SIGNING_JWK": jwt_private_signing_jwk}):
-                setup_login_oauth_client()
-                response = cookies_api.set_logged_in_cookies(self.request, HttpResponse(), self.user)
-                self._copy_cookies_to_request(response, self.request)
-                self.assertTrue(cookies_api.are_logged_in_cookies_set(self.request))
-
-                cookies_api.delete_logged_in_cookies(response)
-                self._copy_cookies_to_request(response, self.request)
-                self.assertFalse(cookies_api.are_logged_in_cookies_set(self.request))
+    def test_delete_and_are_logged_in_cookies_set(self):
```
Comment (iam-mhaseeb) [inline openedx/core/djangoapps/user_authn/tests/test_cookies.py:118]: If we are disabling the `settings_override` middleware then we can avoid this change.

File: openedx/features/course_duration_limits/tests/test_access.py [hunk 10]
```diff
@@ -56,36 +56,34 @@ def test_generate_course_expired_message(self, language, offsets):
         def format_date(date):
             return strftime_localized(date, u'%b %-d, %Y')
 
-        patch_lang = patch('openedx.features.course_duration_limits.access.get_language', return_value=language)
```
Comment (iam-mhaseeb) [inline openedx/features/course_duration_limits/tests/test_access.py:59]: Why did you need to disable language for these factories?

File: openedx/features/discounts/applicability.py [hunk 11]
```diff
@@ -143,7 +143,7 @@ def _is_in_holdback(user):
     """
     Return whether the specified user is in the first-purchase-discount holdback group.
     """
-    if datetime(2020, 8, 1, tzinfo=pytz.UTC) <= datetime.now(tz=pytz.UTC):
+    if datetime(2021, 8, 1, tzinfo=pytz.UTC) <= datetime.now(tz=pytz.UTC):
```
Comment (iam-mhaseeb) [inline openedx/features/discounts/applicability.py:146]: It won't work after 2021. One way is to get the current year and add 1 year into it.

File: openedx/features/discounts/tests/test_applicability.py [hunk 12]
```diff
@@ -171,6 +171,6 @@ def test_holdback_expiry(self):
         with patch('openedx.features.discounts.applicability.stable_bucketing_hash_group', return_value=0):
             with patch(
                 'openedx.features.discounts.applicability.datetime',
-                Mock(now=Mock(return_value=datetime(2020, 8, 1, 0, 1, tzinfo=pytz.UTC)), wraps=datetime),
+                Mock(now=Mock(return_value=datetime(2021, 8, 1, 0, 1, tzinfo=pytz.UTC)), wraps=datetime),
```
Comment (iam-mhaseeb) [inline openedx/features/discounts/tests/test_applicability.py:174]: It won't work after 2021. One way is to get the current year and add 1 year into it.

**Review by iam-mhaseeb — CHANGES_REQUESTED** [review]
Added my feedback

File: lms/templates/instructor/instructor_dashboard_2/membership.html (hunk 8 above)
Comment (zubair-arbi) [reply lms/templates/instructor/instructor_dashboard_2/membership.html:71]: @rehan99000 accidentally removed it and now added it back: https://github.com/edly-io/edx-platform/pull/198#discussion_r565083293

File: openedx/features/course_duration_limits/tests/test_access.py (hunk 10 above)
Comment (rehan99000) [reply openedx/features/course_duration_limits/tests/test_access.py:59]: This was changed in ironwood. in juniper this is removed. I missed it during conflict resolution

File: openedx/features/discounts/applicability.py (hunk 11 above)
Comment (rehan99000) [reply openedx/features/discounts/applicability.py:146]: changed to current year

File: openedx/features/discounts/tests/test_applicability.py (hunk 12 above)
Comment (rehan99000) [reply openedx/features/discounts/tests/test_applicability.py:174]: fixed

File: lms/templates/instructor/instructor_dashboard_2/membership.html (hunk 8 above)
Comment (rehan99000) [reply lms/templates/instructor/instructor_dashboard_2/membership.html:71]: it was a mistake. I have reverted the changes

File: lms/djangoapps/commerce/api/v1/tests/test_views.py (hunk 6 above)
Comment (rehan99000) [reply lms/djangoapps/commerce/api/v1/tests/test_views.py:178]: fixed

**Review by zubair-arbi — APPROVED** [review]
Changes looks good 👍

**Review by taimoor-ahmed-1 — APPROVED** [review]

File: openedx/core/djangoapps/user_authn/tests/test_cookies.py (hunk 9 above)
Comment (rehan99000) [reply openedx/core/djangoapps/user_authn/tests/test_cookies.py:118]: how?

Comment (iam-mhaseeb) [reply openedx/core/djangoapps/user_authn/tests/test_cookies.py:118]: Most probably this test was failing because our settings were getting override but if we are disabling that middleware then default values should apply which should pass this test.

**Review by iam-mhaseeb — APPROVED** [review]

File: openedx/core/djangoapps/user_authn/tests/test_cookies.py (hunk 9 above)
Comment (rehan99000) [reply openedx/core/djangoapps/user_authn/tests/test_cookies.py:118]: This is not our test. this is default `user_authn` cookies tests. This change was also missed in the conflict resolution somehow.

Comment (rehan99000) [reply openedx/core/djangoapps/user_authn/tests/test_cookies.py:118]: also, settings middleware has nothing to do with cookies

File: lms/djangoapps/discussion/django_comment_client/base/tests.py (hunk 7 above)
Comment (rehan99000) [reply lms/djangoapps/discussion/django_comment_client/base/tests.py:1965]: it does but it doesn't affect any test

Comment (rehan99000) [reply lms/djangoapps/discussion/django_comment_client/base/tests.py:1965]: with `pk` provided, it raises User Not found
